python/pacemaker/exitstatus.py

```diff
@@ -4,9 +4,25 @@
 __copyright__ = "Copyright 2023-2024 the Pacemaker project contributors"
 __license__ = "GNU Lesser General Public License version 2.1 or later (LGPLv2.1+)"
 
+import ctypes
+from ctypes.util import find_library
 from enum import IntEnum, unique
 
 
+def _load_library(basename):
+    """Find and load the library with the given base name, or None if not found."""
+    path = find_library(basename)
+
+    if path is None:
+        return None
+
+    return ctypes.cdll.LoadLibrary(path)
+
+
+_libcrmcommon = _load_library("crmcommon")
+_libcrmcommon.crm_exit_str.restype = ctypes.c_char_p
```

Comment:
I've tried to see if there's any best practices python recommends for this, without much luck.  If you grep around in /usr/lib*/python*/, you'll find that modules do all sorts of things.  Some just have a big list of these sorts of lines in a file somewhere, some do it in as part of initializing an object.  So I think we could basically do whatever we feel is best.

Reply:
This might be a good direction:
```
diff --git a/python/pacemaker/_library.py b/python/pacemaker/_library.py
index b9e492341f..cd6bc4ef44 100644
--- a/python/pacemaker/_library.py
+++ b/python/pacemaker/_library.py
@@ -1,6 +1,6 @@
 """A module providing private library management code."""
 
-__all__ = ["load_library"]
+__all__ = ["_libcrmcommon"]
 __copyright__ = "Copyright 2024 the Pacemaker project contributors"
 __license__ = "GNU Lesser General Public License version 2.1 or later (LGPLv2.1+)"
 
@@ -32,3 +32,7 @@ def load_library(basename):
         raise FileNotFoundError(basename)
 
     return ctypes.cdll.LoadLibrary(path)
+
+
+_libcrmcommon = load_library("crmcommon")
+_libcrmcommon.crm_exit_str.restype = ctypes.c_char_p
diff --git a/python/pacemaker/exitstatus.py b/python/pacemaker/exitstatus.py
index c699e2599e..03f7d2c8e2 100644
--- a/python/pacemaker/exitstatus.py
+++ b/python/pacemaker/exitstatus.py
@@ -4,14 +4,9 @@ __all__ = ["ExitStatus"]
 __copyright__ = "Copyright 2023-2024 the Pacemaker project contributors"
 __license__ = "GNU Lesser General Public License version 2.1 or later (LGPLv2.1+)"
 
-import ctypes
 from enum import IntEnum, unique
 
-from pacemaker._library import load_library
-
-
-_libcrmcommon = load_library("crmcommon")
-_libcrmcommon.crm_exit_str.restype = ctypes.c_char_p
+from pacemaker._library import _libcrmcommon
 
 
 # These values must be kept in sync with include/crm/common/results.h
```

Putting this stuff into `__init__.py` leads to circular import weirdness.

Reply:
looks reasonable